googledataprocauthenticator/google.py

```diff
@@ -0,0 +1,271 @@
+
+    Returns:
+        Sequence[str]: each key is a str of one of the users credentialed accounts
+
+    Raises:
+        sparkmagic.livyclientlib.BadUserConfigurationException: if gcloud cannot be invoked
+    """
+    accounts_json = ""
+    if os.name == "nt":
+        command = _CLOUD_SDK_WINDOWS_COMMAND
+    else:
+        command = _CLOUD_SDK_POSIX_COMMAND
+    try:
+        command = (command,) + _CLOUD_SDK_USER_CREDENTIALED_ACCOUNTS_COMMAND
+        # run `gcloud auth list` command 
+        accounts_json = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        all_accounts = json.loads(accounts_json)
+        credentialed_accounts = set()
+        active_account = None
+        for account in all_accounts:
+            try:
+                _cloud_sdk.get_auth_access_token(account['account'])
+                if account['status'] == 'ACTIVE':
+                    active_account = account['account']
+                credentialed_accounts.add(account['account'])
+            # when`gcloud auth print-access-token --account=account` fails we don't add it to
+            # credentialed_accounts dict that populates account dropdown widget
+            except UserAccessTokenError:
+                pass
+        return credentialed_accounts, active_account
+    except (subprocess.CalledProcessError, OSError, IOError) as caught_exc:
+        new_exc = BadUserConfigurationException("Gcloud cannot be invoked.")
+        raise new_exc from caught_exc
+
+def get_credentials_for_account(account, scopes_list):
+    """Load all of user's credentialed accounts with ``gcloud auth describe ACCOUNT`` command.
+
+    Args:
+        account (str): user credentialed account to return credentials for
+        scopes_list (Sequence[str]): list of scopes to include in the credentials.
+    
+    Returns:
+        google.oauth2.credentials.Credentials: The constructed credentials
+
+    Raises:
+        ValueError: If `gcloud auth describe ACCOUNT --format json` returns json not in the expected format.
+        google.auth.exceptions.UserAccessTokenError: if credentials could not be found for the given account. 
+    """
+    if os.name == "nt":
+        command = _CLOUD_SDK_WINDOWS_COMMAND
+    else:
+        command = _CLOUD_SDK_POSIX_COMMAND
+    try:
+        describe_account_command = ("auth", "describe", account, '--format', 'json')
+        command = (command,) + describe_account_command
+        account_json = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        account_describe = json.loads(account_json)
+        return Credentials.from_authorized_user_info(account_describe, scopes=scopes_list)
+    # subprocess.CalledProcessError, OSError, IOError are from subprocess
+    # ValueError is from Credentials.from_authorized_user_info
+    # JSONDecodeError is from json.loads.
+    except (subprocess.CalledProcessError, OSError, IOError, ValueError, json.JSONDecodeError) as caught_exc:
```

Comment:
I'll delete these comments after, but I was just wondering if I can instead of this line do `except Exception as caught_exc:` (pylint complains when I do this) or is it better to just list all of them out?

Reply:
Since we're just wrapping and reraising the exception, it's fine to have a more general catch here. You can tell Pylint to ignore an exception on a specific line with a comment https://stackoverflow.com/questions/28829236/is-it-possible-to-ignore-one-single-specific-line-with-pylint